packages/runtime/src/useAsyncResultManager.ts

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useRef, useState } from "react";
+import { isObservable, isPromise, Observable, Subscription } from "./utils";
+
+export const useAsyncResultManager = <TResult>(
+  result: TResult | Promise<TResult> | Observable<TResult>,
+  cachedResult: TResult,
+): TResult | null => {
+  const observerRef = useRef<Subscription | null>(null);
+  const [subscriptionResult, setSubscriptionResult] = useState<TResult | null>(null);
+  const [promiseResult, setPromiseResult] = useState<TResult | null>(null);
+
+  useEffect(() => {
+    let isMounted = true;
+
+    if (isObservable(result)) {
+      observerRef.current = result.subscribe({
+        next: (value) => {
+          if (!isMounted) return;
+          setSubscriptionResult(value);
+        },
+        error: () => {
+          if (!isMounted) return;
+          setSubscriptionResult(null);
```

Comment:
What's the point of  the `error` function?
Currently, it behaves the same way as calling `next(null)`, what's the motivation to call it?

Also, the client code can handle the errors by standard means. Maybe if we want to follow **The Single Responsibility Principle** then error handling shouldn't be the concern of the plugin.